modules/liquidity-pools/margin/src/lib.rs

```diff
@@ -192,6 +195,9 @@ decl_module! {
 			let who = ensure_signed(origin)?;
 			ensure!(Self::is_owner(pool_id, &who), Error::<T>::NoPermission);
 			ensure!(Self::enabled_trading_pair(&pair).is_some(), Error::<T>::TradingPairNotEnabled);
+
+			<T::OnEnableTradingPair as OnEnableTradingPair>::ensure_can_enable_trading_pair(pool_id, pair)?;
```

Comment:
Liquidity pools manager is designed to do this kind of checks. I think we had discussed on slack and agreed adding a `MarginProtocolLiquidityPoolsManager` trait to hold the trading pair check. Any particular reason to use a trait type instead?

Note `OnXXX` is for when `XXX` is done, to notice who implement it to do the clean up, it doesn't make sense to me to use it as a guard function.